sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -0,0 +1,460 @@
+            parsed_url.scheme,
+            parsed_url.netloc.rstrip('/'),
+            account_path,
+            parsed_url.query)
+        table_name = unquote(table_path[-1])
+        if not table_name:
+            raise ValueError("Invalid URL. Please provide a URL with a valid table name")
+        return cls(account_url, table_name=table_name, credential=credential, **kwargs)
+
+    @distributed_trace
+    def get_table_access_policy(
+            self,
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> dict[str,AccessPolicy]
+        """Retrieves details about any stored access policies specified on the table that may be
+        used with Shared Access Signatures.
+
+        :return: Dictionary of SignedIdentifiers
+        :rtype: dict[str,AccessPolicy]
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        timeout = kwargs.pop('timeout', None)
+        try:
+            _, identifiers = self._client.table.get_access_policy(
+                table=self.table_name,
+                timeout=timeout,
+                cls=return_headers_and_deserialized,
+                **kwargs)
+        except HttpResponseError as error:
+            process_table_error(error)
+        return {s.id: s.access_policy or AccessPolicy() for s in identifiers}
+
+    @distributed_trace
+    def set_table_access_policy(
+            self,
+            signed_identifiers,  # type: dict[str,AccessPolicy]
+            **kwargs):
+        # type: (...) -> None
+        """Sets stored access policies for the table that may be used with Shared Access Signatures.
+
+        :param signed_identifiers:
+        :type signed_identifiers: dict[str,AccessPolicy]
+        :return: None
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        self._validate_signed_identifiers(signed_identifiers)
+        identifiers = []
+        for key, value in signed_identifiers.items():
+            if value:
+                value.start = serialize_iso(value.start)
+                value.expiry = serialize_iso(value.expiry)
+            identifiers.append(SignedIdentifier(id=key, access_policy=value))
+        signed_identifiers = identifiers  # type: ignore
+        try:
+            self._client.table.set_access_policy(
+                table=self.table_name,
+                table_acl=signed_identifiers or None,
+                **kwargs)
+        except HttpResponseError as error:
+            process_table_error(error)
+
+    @distributed_trace
+    def create_table(
+            self,
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> str
+        """Creates a new table under the current account.
+
+        :return: Table created
+        :rtype: str
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        table_properties = TableProperties(table_name=self.table_name, **kwargs)
+        table = self._client.table.create(table_properties)
```

Comment:
This should be wrapped in the `try/except HttpResponseError` handling.

Reply:
This comment applies to every time we call the API (i.e. every time we call into `self._client`) - so there's a couple more below and I wont comment on all of them :)